src/main/java/com/spartronics4915/frc2019/auto/modes/CharacterizeDrive.java

```diff
@@ -5,24 +5,55 @@
 import com.spartronics4915.frc2019.auto.actions.CollectAccelerationData;
 import com.spartronics4915.frc2019.auto.actions.CollectVelocityData;
 import com.spartronics4915.frc2019.auto.actions.WaitAction;
+import com.spartronics4915.frc2019.subsystems.Drive;
 import com.spartronics4915.lib.physics.DriveCharacterization;
 import com.spartronics4915.lib.util.Logger;
 
 import java.util.ArrayList;
 import java.util.List;
 
-public class CharacterizeHighGearStraight extends AutoModeBase
+public class CharacterizeDrive extends AutoModeBase
 {
 
+    public enum SideToCharacterize
+    {
+        BOTH, LEFT, RIGHT;
+
+        public double getVelocitiesFromDrive(Drive drive)
+        {
+            switch (this)
+            {
+                case BOTH:
+                    return
+                        Math.abs(drive.getLeftVelocityTicksPer100ms()) +
+                        Math.abs(drive.getRightVelocityTicksPer100ms());
+                case LEFT:
+                    return drive.getLeftVelocityTicksPer100ms();
```

Comment:
should these have an abs?  if not, would you expect us to characterize differently for fwd and reverse?

Reply:
Whoops. That's a mistake.